pkg/image/registryclient/client_mirrored.go

```diff
@@ -216,6 +216,9 @@ func (r *blobMirroredRepository) alternates(ctx context.Context, fn func(r Repos
 		if err != nil {
 			return err
 		}
+		if len(alternates) == 0 {
```

Comment:
@smarterclayton alternatively we could check for nil strategy in https://github.com/openshift/library-go/blob/master/pkg/image/registryclient/client_mirrored.go#L207 similarly to [initialRepos](https://github.com/openshift/library-go/blob/master/pkg/image/registryclient/client_mirrored.go#L122) or [errorRepos](https://github.com/openshift/library-go/blob/master/pkg/image/registryclient/client_mirrored.go#L145)